controllers/openstackdeploy_controller.go

```diff
@@ -106,6 +107,12 @@ func (r *OpenStackDeployReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		)
 	}
 
+	uuid, _ := exec.Command("uuidgen").Output()
+
+	if instance.Spec.DeployIndentifier == "" {
```

Comment:
defaults are set in the webhook https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/api/v1beta1/openstackdeploy_webhook.go#L49

Reply:
I expect modifying the spec here would create an infinite loop.

Reply:
I've not tested this yet. Will see how it behaves.